app/Services/Auth/OauthLoginService.php

```
@@ -0,0 +1,141 @@
+    public function login(string $provider, object $oauthUser, OauthSetting $oauthSetting): User
+    {
+        $email = strtolower(trim((string) $oauthUser->email));
+        if ($email === '') {
+            throw new HttpException(403, 'OAuth provider did not return an email address');
+        }
+
+        $user = $provider === 'oidc'
+            ? $this->resolveOidcUser($oauthUser, $oauthSetting, $email)
+            : $this->resolveOauthUser($oauthUser, $oauthSetting, $email);
+
+        Auth::login($user);
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\Illuminate\Support\Facades\Auth' in method 'login'. (undefined)

(StaticAccess)